WalletWasabi/Wallets/Wallet.cs

```diff
@@ -177,7 +179,7 @@ public List<TransactionSummary> BuildHistorySummary(bool sortForUI = false)
 			}
 			else
 			{
-				mapByTxid.Add(coin.TransactionId, new TransactionSummary(coin.Transaction, coin.Amount));
+				mapByTxid.Add(coin.TransactionId, new TransactionSummary(coin.Transaction, coin.Amount, feeFetcher: () => coin.Transaction.GetFee() ?? TransactionFeeProvider.GetFee(coin.TransactionId)));
```

Comment:
My idea here was to try to calculate the fee on client-side.

If we can => we are happy, we don't need the `TransactionFeeProvider`
If we cannot => fetch it from the `TransactionFeeProvider`